### PR TITLE
Migrating to go module and go 1.14

### PR DESCRIPTION
**Pull request by jpbelangerupgrade**

### CODE REVIEW

**Comment by devopsciupgrade** [comment]
Dockerfile Twistlock vulnerabilities and compliance scan:
```

Vulnerabilities
---------------
Image    ID    CVE    Package    Version    Severity    Status    CVSS    Grace days
-----    --    ---    -------    -------    --------    ------    ----    ----------
Vulnerability threshold check results: PASS

Compliance
----------
Image                          ID                  Severity    Description
-----                          --                  --------    -----------
s3-sftp-proxy:1590679035226    685770e027d383b6    high        (CIS_Docker_CE_v1.1.0 - 4.1) Image should be created with a non-root user
Compliance threshold check results: PASS

```

[For more details check the Twistlock Console using the temporary tag 's3-sftp-proxy:1590679035226'](https://credify.atlassian.net/wiki/spaces/TOP/pages/1245809526/Docker+vulnerability+management)

File: Dockerfile [hunk 1]
```diff
@@ -1,20 +1,42 @@
...
+    && go install \
+    && echo
+
+# Definitive image
+FROM docker-upgrade.artifactory.build.upgrade.com/container-base:2.0.20200406.0-205
 
-FROM alpine:3.8
+# We know it's bad but this is for docker socket to be accessible when mounted used with twistcli actually
+# ticket open for that issue: https://credify.atlassian.net/browse/TOP-9238
+USER root
```
Comment (gjacquetupgrade) [inline Dockerfile]: does it have to be root?
is it a copy paste as the comment suggest or is it to allow binding to port 22.
If so we may eventually consider binding to 2222 and export 22 on the service / ingress side

Comment (jpbelangerupgrade) [reply Dockerfile]: removed it already

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,19 @@
+module github.com/moriyoshi/s3-sftp-proxy
+
+go 1.13
```
Comment (gjacquetupgrade) [inline go.mod:3]: we go all the way to 1.14 now!

File: Dockerfile (hunk 1 above)
Comment (jpbelangerupgrade) [reply Dockerfile]: default is `_bind = ":10022"`

**Pull request merged**